### PR TITLE
refactor: enable time travel by default

### PR DESCRIPTION
**Pull request by zwang28**
I hereby agree to the terms of the [RisingWave Labs, Inc. Contributor License Agreement](https://gist.github.com/TennyZhuang/f00be7f16996ea48effb049aa7be4d66#file-rw_cla).

## What's changed and what's your intention?

This PR modifies several configurations to enable time travel by default, ensuring regular batch query remains functional even if there is a missing recent version in compute node.
- `min_sst_retention_time_sec` is reduced from 86400 to 10800 (3 hour). Since after enabling time travel, the deletion of SSTs is entirely dependent on full GC. A smaller `min_sst_retention_time_sec` permits the earlier deletion of SST. However it cannot be too small, because it also determines the maximum permitted [barrier latency](https://github.com/risingwavelabs/risingwave/pull/18757) and [compaction latency](https://github.com/risingwavelabs/risingwave/pull/18779).
- `full_gc_interval_sec` is reduced from 86400 to 600, aiming to trigger full gc more frequently.
-  The system param `time_travel_retention_ms` is increased from 0 to 600000 (10 minutes). 
    - Since `min_sst_retention_time_sec` will keep recent SSTs anyway, `time_travel_retention_ms` can actually be as large as `min_sst_retention_time_sec` without introducing object store overhead. However it will introduce meta store overhead.
    - Later PR will limit the minimum value of `min_sst_retention_time_sec`.

A new configuration `time_travel_version_cache_capacity` is added.

## Checklist

- [ ] I have written necessary rustdoc comments
- [ ] I have added necessary unit tests and integration tests
- [ ] I have added test labels as necessary. See [details](https://github.com/risingwavelabs/risingwave/blob/main/docs/developer-guide.md#ci-labels-guide).
- [ ] I have added fuzzing tests or opened an issue to track them. (Optional, recommended for new SQL features #7934).
- [ ] My PR contains breaking changes. (If it deprecates some features, please create a tracking issue to remove them in the future).
- [ ] All checks passed in `./risedev check` (or alias, `./risedev c`)
- [ ] My PR changes performance-critical code. (Please run macro/micro-benchmarks and show the results.)

- [ ] My PR contains critical fixes that are necessary to be merged into the latest release. (Please check out the [details](https://github.com/risingwavelabs/risingwave/blob/main/CONTRIBUTING.md))

## Documentation

- [ ] My PR needs documentation updates. (Please use the **Release note** section below to summarize the impact on users)

## Release note

If this PR includes changes that directly affect users or other significant modifications relevant to the community, kindly draft a release note to provide a concise summary of these changes. Please prioritize highlighting the impact these changes will have on users.

</details>

### CODE REVIEW

**Comment by zwang28** [comment]
This PR can only be merged after etcd is deprecated.

File: src/common/src/system_param/mod.rs [hunk 1]
```diff
@@ -91,7 +91,7 @@ macro_rules! for_all_params {
             { enable_tracing,                           bool,                           Some(false),                    true,   "Whether to enable distributed tracing.", },
             { use_new_object_prefix_strategy,           bool,                           None,                           false,  "Whether to split object prefix.", },
             { license_key,                              risingwave_license::LicenseKey, Some(Default::default()),       true,   "The license key to activate enterprise features.", },
-            { time_travel_retention_ms,                 u64,                            Some(0_u64),                    true,   "The data retention period for time travel, where 0 indicates that it's disabled.", },
+            { time_travel_retention_ms,                 u64,                            Some(600000_u64),              true,   "The data retention period for time travel.", },
```
Comment (hzxa21) [inline src/common/src/system_param/mod.rs:94]: IIUC, old cluster won't use the new default value after upgrade. I think we need to have logic to overwrite 0 `time_travel_retention_ms` to the new default forcibly.

**Comment by hzxa21** [comment]
> With min_sst_retention_time_sec=3600, the maximum permitted barrier latency is 1 hour.

Given that SST sync is triggered after collecting barriers, the main issue here is for spilled SSTs, which can happen at any time before barrier collection. In other words, if a spilling happens 1hour before the barrier is completed, commit_epoch will fail. Otherwise if the streaming pipeline is stuck by compaction lag or backpressure with no spilling happens, commit_epoch can still succeed even if barrier latency is high.

However, we do see in production that when join amp is high for some use cases, spilling can happen and barrier latency can go higher than 1 hour occasionally. Prior to this PR, the occasional spilling won't be an issue but after this PR, I believe their clusters will enter a recovery loop, which makes me a little bit concerned. 

Brainstorm idea: now that we have [bidi-streaming RPC](https://github.com/risingwavelabs/risingwave/blob/71001d8444b7176fd2f406eecc65c8e516f0d4e6/proto/stream_service.proto#L90) for barrier injection/collection and the spilled SST information is maintained in a single place in [`UploaderData`](https://github.com/risingwavelabs/risingwave/blob/71001d8444b7176fd2f406eecc65c8e516f0d4e6/src/storage/src/hummock/event_handler/uploader/mod.rs#L1081). I am thinking we can simply mark the spilled SST illegible for GC in-memory (reset on recovery). Not sure whether there are other complexities here.

**Comment by zwang28** [comment]
> Brainstorm idea: now that we have [bidi-streaming RPC](https://github.com/risingwavelabs/risingwave/blob/71001d8444b7176fd2f406eecc65c8e516f0d4e6/proto/stream_service.proto#L90) for barrier injection/collection and the spilled SST information is maintained in a single place in [UploaderData](https://github.com/risingwavelabs/risingwave/blob/71001d8444b7176fd2f406eecc65c8e516f0d4e6/src/storage/src/hummock/event_handler/uploader/mod.rs#L1081). I am thinking we can simply mark the spilled SST illegible for GC in-memory (reset on recovery)

Sounds good to me. The marked SSTs are filtered out for both GC and timestamp verification of commit_epoch.

Before that is ready, what about increase min_sst_retention_time_sec to 3 hours? Note that it can still be modified on demand for exisiting clusters, when larger barrier latency is expected.

**Comment by hzxa21** [comment]
> > Brainstorm idea: now that we have [bidi-streaming RPC](https://github.com/risingwavelabs/risingwave/blob/71001d8444b7176fd2f406eecc65c8e516f0d4e6/proto/stream_service.proto#L90) for barrier injection/collection and the spilled SST information is maintained in a single place in [UploaderData](https://github.com/risingwavelabs/risingwave/blob/71001d8444b7176fd2f406eecc65c8e516f0d4e6/src/storage/src/hummock/event_handler/uploader/mod.rs#L1081). I am thinking we can simply mark the spilled SST illegible for GC in-memory (reset on recovery)
> 
> Sounds good to me. The marked SSTs are filtered out for both GC and timestamp verification of commit_epoch. I'll work on it.
> 
> Before that is ready, what about increasing min_sst_retention_time_sec to 3 hours? Note that it can still be modified on demand for exisiting clusters, when larger barrier latency is expected.

SGTM

**Comment by hzxa21** [comment]
> This PR can only be merged after etcd is deprecated.

The [time_travel_enabled](https://github.com/risingwavelabs/risingwave/blob/f14897fe543fd16551dc06b6ef386d9cb09bb81a/src/meta/src/hummock/manager/time_travel.rs#L58) check will ignore `time_travel_retention_ms` if the meta backend is etcd. Does it mean that this PR should not be blocked by etcd deprecation?

**Review by hzxa21 — APPROVED** [review]

**Pull request merged**